crates/networking/src/gossip.rs

```diff
@@ -321,48 +325,21 @@ pub struct GossipMessage {
 #[derive(Serialize, Deserialize, Debug)]
 pub enum MyBehaviourRequest {
     Handshake {
-        ecdsa_public_key: K256VerifyingKey,
-        signature: K256Signature,
+        crypto_public_key: CryptoPublicKey,
+        signature: CryptoSignature,
     },
     Message {
         topic: String,
         raw_payload: Vec<u8>,
     },
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
-pub struct EcdsaSignatureWrapper(pub [u8; 65]);
-
-impl serde::Serialize for EcdsaSignatureWrapper {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
-        serializer.serialize_bytes(&self.0)
-    }
-}
-
-impl<'de> serde::Deserialize<'de> for EcdsaSignatureWrapper {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        let bytes = <Vec<u8>>::deserialize(deserializer)?;
-        if bytes.len() != 65 {
-            return Err(serde::de::Error::custom("Invalid signature length"));
-        }
-        let mut signature = [0u8; 65];
-        signature.copy_from_slice(&bytes);
-        Ok(EcdsaSignatureWrapper(signature))
-    }
-}
-
 #[non_exhaustive]
 #[derive(Serialize, Deserialize, Debug)]
 pub enum MyBehaviourResponse {
     Handshaked {
-        ecdsa_public_key: K256VerifyingKey,
-        ecdsa_signature: K256Signature,
+        crypto_public_key: CryptoPublicKey,
```

Comment:
I feel that the `crypto` prefix is meaningless. We should find a better one. Is this key used to just handshake or also to sign gossip/networking messages? We can come up with a better name or just drop this prefix to be `public_key` and `signature`.